public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,337 @@
+  }, [confirmModalState.isRequestingToClose, isLoading]);
+
+  useEffect(() => {
+    getDetectors();
+  }, []);
+
+  // Handles all changes in the assoicated detectors such as unlinking or new detectors associated
+  useEffect(() => {
+    // Gets all augmented saved objects
+    savedObjectLoader
+      .findAll()
+      .then((resp: any) => {
+        if (resp != undefined) {
+          const savedAugmentObjectsArr: ISavedAugmentVis[] = get(
+            resp,
+            'hits',
+            []
+          );
+          const curSelectedDetectors = getAssociatedDetectors(
+            Object.values(allDetectors),
+            savedAugmentObjectsArr
+          );
+          setSelectedDetectors(curSelectedDetectors);
+          setIsLoadingFinalDetectors(false);
+        }
+      })
+      .catch((error) => {
+        core.notifications.toasts.addDanger(
+          prettifyErrorMessage(`Unable to fetch associated detectors: ${error}`)
+        );
+      });
+  }, [allDetectors]);
+
+  // cross checks all the detectors that exist with all the savedAugment Objects to only display ones
+  // that are associated to the current visualization
+  const getAssociatedDetectors = (
+    detectors: DetectorListItem[],
+    savedAugmentObjects: ISavedAugmentVis[]
+  ) => {
+    // Filter all savedAugmentObjects that aren't linked to the specific visualization
+    const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+      savedAugmentObjects.filter(
+        (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+      );
+
+    // Map all detector IDs for all the found augmented vis objects
+    const savedAugmentDetectorsSet = new Set(
+      savedAugmentForThisVisualization.map((savedObject) =>
+        get(savedObject, 'pluginResourceId', '')
+      )
+    );
+
+    // filter out any detectors that aren't on the set of detectors IDs from the augmented vis objects.
+    const detectorsToDisplay = detectors.filter((detector) =>
+      savedAugmentDetectorsSet.has(detector.id)
+    );
+    return detectorsToDisplay;
+  };
+
+  const onUnlinkDetector = async () => {
+    setIsLoadingFinalDetectors(true);
+    await savedObjectLoader.findAll().then(async (resp: any) => {
+      if (resp != undefined) {
+        const savedAugmentObjects: ISavedAugmentVis[] = get(resp, 'hits', []);
+        // gets all the saved object for this visualization
+        const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+          savedAugmentObjects.filter(
+            (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+          );
+
+        // find saved Augment object matching detector we want to unlink
+        // There should only be one detector and vis pairing
+        const savedAugmentToUnlink = savedAugmentForThisVisualization.find(
+          (savedObject) =>
+            get(savedObject, 'pluginResourceId', '') === detectorToUnlink.id
+        );
+        const savedObjectToUnlinkId = get(savedAugmentToUnlink, 'id', '');
```

Comment:
nit: you could combine all of these into a chained set of cmds to simplify it and to remove having to keep intermediate consts.

Reply:
I can delete savedAugmentObjects and savedObjectToUnlinkId. Wanted to keep 2 constants for the filter operations to keep them more readable rather then nesting the gets and filters too much.